src/main/java/net/countercraft/movecraft/combat/features/tracking/BlockBehaviorOverride.java

```diff
@@ -0,0 +1,420 @@
+            if (this.magicNumbers == null) {
+                this.magicNumbers = this.getCraftMagicNumbersClass();
+            }
+            Method method = magicNumbers.getMethod("getBlock", Material.class);
+            return method.invoke(null, m);
+        }
+
+        protected static <T> void writeField(@NotNull Object block, @NotNull Consumer<T> whatToDoWithField, String fieldName) throws IllegalAccessException, NoSuchFieldException, ClassCastException {
+            Field field = FieldUtils.getField(block.getClass(), fieldName, true);
+            T obj = (T)field.get(block);
+            whatToDoWithField.accept(obj);
+        }
+
+        protected static <T> void writeField(@NotNull Object block, T value, String fieldName) throws IllegalAccessException, NoSuchFieldException, ClassCastException {
+            Field field = FieldUtils.getField(block.getClass(), fieldName, true);
+            T obj = (T)field.get(block);
+            field.set(block, value);
+        }
+
+        protected static <T> Optional<T> getFieldValueSafe(@NotNull Object instance, String fieldName) {
+            try {
+                return Optional.ofNullable(getFieldValue(instance, fieldName));
+            } catch(Exception ex) {
+                return Optional.empty();
+            }
+        }
+        protected static <T> T getFieldValue(@NotNull Object instance, String fieldName) throws IllegalAccessException, NoSuchFieldException, ClassCastException {
+            Field field = FieldUtils.getField(instance.getClass(), fieldName, true);
+            T obj = (T)field.get(instance);
+            return obj;
+        }
+    }
+
+    private static class NMSSpigotMappings extends NMSHelper {
+
+        private static final String FIELD_NAME_IGNITE_ODDS = "O";
+        private static final String FIELD_NAME_BURN_ODDS = "P";
+        private static final String FIELD_NAME_BLAST_RESISTANCE = "aH";
+
+        public NMSSpigotMappings() {
+            this(FIELD_NAME_BLAST_RESISTANCE, FIELD_NAME_BURN_ODDS, FIELD_NAME_IGNITE_ODDS);
+        }
+
+        protected NMSSpigotMappings(String blastResField, String burnOddsField, String igniteOddsField) {
+            super(blastResField, burnOddsField, igniteOddsField);
+        }
+
+        @Override
+        Class<?> getCraftMagicNumbersClass() throws ClassNotFoundException {
+            String packageName = Bukkit.getServer().getClass().getPackage().getName();
+            String version = packageName.substring(packageName.lastIndexOf('.') + 1);
+            return Class.forName("org.bukkit.craftbukkit." + version + ".util.CraftMagicNumbers");
+        }
+    }
+
+    private static class NMSMojangMappings extends NMSHelper {
+
+        private static final String FIELD_NAME_IGNITE_ODDS = "igniteOdds";
+        private static final String FIELD_NAME_BURN_ODDS = "burnOdds";
+        private static final String FIELD_NAME_BLAST_RESISTANCE = "explosionResistance";
```

Comment:
Similar here, a comment with tested versions (1.20.6 and 1.21) would be good.

Reply:
Can do tests later